sc/social/like/utils.py

```diff
@@ -9,51 +9,40 @@
 
 def get_images_view(context):
     view = context.unrestrictedTraverse('@@images', None)
-    field = 'image'
-    if view:
-        fields = ['image', 'leadImage', 'portrait']
-        if IBaseContent.providedBy(context):
-            schema = context.Schema()
-            field = [f for f in schema.keys() if f in fields]
-            if field:
-                field = field[0]
-                # if a content has an image field that isn't an ImageField
-                # (for example a relation field), set field='' to avoid errors
-                if schema[field].type not in ['image', 'blob']:
-                    field = ''
-    return (view, field) if (view and field) else (None, None)
+    if view is None:
+        return (None, None)
+    if not IBaseContent.providedBy(context):
+        return (view, 'image')
+    fields = ['image', 'leadImage', 'portrait']
+    schema = context.Schema()
+    field = [f for f in schema.keys() if f in fields]
+    if len(field) == 0:
+        return (None, None)
+    field = field[0]
+    # if a content has an image field that isn't an ImageField (for example a relation field)
+    if schema[field].type not in ['image', 'blob']:
+        return (None, None)
+    return (view, field)
 
 
-def get_content_image(context,
-                      scale='large',
-                      width=None,
-                      height=None):
+def get_content_image(context):
+    """Return image object just if we can get image size."""
     view, field = get_images_view(context)
-    img = None
-    if view:
-        try:
-            sizes = view.getImageSize(field)
-        except AttributeError:
-            sizes = img = None
-        if sizes == (0, 0) or sizes == ('', ''):
-            # this avoid strange cases where we can't get size infos.
-            # for example if the loaded image in a news is a bmp or a tiff
-            return None
-        if sizes:
-            kwargs = {}
-            if not (width or height):
-                kwargs['scale'] = scale
-            else:
-                new = (width, height)
-                width, height = _image_size(sizes, new)
-                kwargs['width'] = width
-                kwargs['height'] = height
-                kwargs['direction'] = 'down'
-            try:
-                img = view.scale(fieldname=field, **kwargs)
-            except (AttributeError, TypeError):
-                img = None
-    return img
+    if view is None:
+        return
+    # XXX: If I recall correctly, Dexterity works different to do this
```

Comment:
Yeah, that's right: https://github.com/collective/collective.nitf/blob/master/src/collective/nitf/browser/scaling.py#L55-L58